libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DHCP.c

```diff
@@ -552,31 +568,30 @@ BaseType_t xReturn;
 TickType_t xTimeoutTime = ( TickType_t ) 0;
 
 	/* Create the socket, if it has not already been created. */
-	if( xDHCPData.xDHCPSocket == NULL )
+	if( xDHCPSocket == NULL )
 	{
-		xDHCPData.xDHCPSocket = FreeRTOS_socket( FREERTOS_AF_INET, FREERTOS_SOCK_DGRAM, FREERTOS_IPPROTO_UDP );
-		if( xDHCPData.xDHCPSocket != FREERTOS_INVALID_SOCKET )
+		xDHCPSocket = FreeRTOS_socket( FREERTOS_AF_INET, FREERTOS_SOCK_DGRAM, FREERTOS_IPPROTO_UDP );
+		if( xDHCPSocket != FREERTOS_INVALID_SOCKET )
 		{
 
 			/* Ensure the Rx and Tx timeouts are zero as the DHCP executes in the
 			context of the IP task. */
-			FreeRTOS_setsockopt( xDHCPData.xDHCPSocket, 0, FREERTOS_SO_RCVTIMEO, ( void * ) &xTimeoutTime, sizeof( TickType_t ) );
-			FreeRTOS_setsockopt( xDHCPData.xDHCPSocket, 0, FREERTOS_SO_SNDTIMEO, ( void * ) &xTimeoutTime, sizeof( TickType_t ) );
+			( void ) FreeRTOS_setsockopt( xDHCPSocket, 0, FREERTOS_SO_RCVTIMEO, &( xTimeoutTime ), sizeof( TickType_t ) );
+			( void ) FreeRTOS_setsockopt( xDHCPSocket, 0, FREERTOS_SO_SNDTIMEO, &( xTimeoutTime ), sizeof( TickType_t ) );
```

Comment:
Should we be checking the return value of `FreeRTOS_setsockopt`?

Reply:
In fact there is nothing dynamic that can go wrong, there are no malloc's needed or so.

Reply:
I see. THis is just a way to stick to the Berkley standard of APIs then, correct?